### PR TITLE
SPORTSHUB-227 Add-exception-handling-if-google-api-does-not-load

### PR DESCRIPTION
**Pull request by inzaenity**
Changed The Autocomplete so that you must choose an autocomplete option (just writing down an address without selecting one from the dropdown will automatically clear itself)

Added an Alert for that page which can be triggered by setting hasAlert to be true and then setting AlertMessage to Whatever

If Location is somehow "" on that page submit, It will just show an alert which says location not found and not let you click next (didn't really change the existing error checking using hasError written by someone else)

### CODE REVIEW

**Review by 41dan — APPROVED** [review]
yep thanks danny looks a lot more robust

File: frontend/app/(footer)/event/create/page.tsx [hunk 1]
```diff
@@ -197,7 +230,14 @@ export default function CreateEvent() {
             </div>
```
Comment (ReggieStar11) [inline frontend/app/(footer)/event/create/page.tsx:230]: unneccassry line

File: frontend/components/utility/AutoComplete.tsx [hunk 2]
```diff
@@ -27,17 +35,31 @@ const LocationAutocompleteForm: React.FC<AutocompleteFormProps> = ({ location, u
...
+        setSelectionMade(true);
         updateField({ location: place.formatted_address });
+
         try {
           const { lat, long } = await validateLocation(place.formatted_address);
           updateField({ lat, long });
+          setHasError(false);
+          setLocationError("");
         } catch (error) {
           console.error(error);
```
Comment (ReggieStar11) [inline frontend/components/utility/AutoComplete.tsx:47]: shouldnt we use a logger instead of console.error?

**Review by ReggieStar11 — COMMENTED** [review]
fix the two changes then lgtm

**Review by owya490 — APPROVED** [review]
thanks danny

**Review by owya490 — APPROVED** [review]
nice

**Pull request merged**